exercise/protocol/aplus.py

```diff
@@ -111,6 +113,8 @@ def load_feedback_page(request, url, exercise, submission, no_penalties=False):
                 exercise.service_url)
             page.errors.append(_('ASSESSMENT_SERVICE_ERROR_RESPONDED_ERROR'))
         submission.save()
+        if submission.meta_data.get('lti-launch-id') != None:
+            send_lti_points(request, submission)
```

Comment:
If the exercise is asynchronously graded, then there are no points at this time yet. This sends zero points to the LTI Platform/Moodle at this time. Is this intentional?

Reply:
No, this was only intended to be a way of sending results of synchronously graded exercises, as async results are handled in `exercise/async_views.py`. What would be the proper place of handling synchronous results?

Reply:
This file is the correct place to handle synchronous exercises (questionnaires). You could just move the LTI part (`send_lti_points()`) to the `if` branches so that `send_lti_points()` is only called here when the points really are ready. Though, it is better to save the submission first (`submission.save()`). You could just add more `if` conditions like

```python
if  page.is_graded and page.is_save() and submission.meta_data.get('lti-launch-id'):
    send_lti_points(request, submission)
```